llvm/test/Transforms/InstCombine/AMDGPU/amdgcn-intrinsics.ll

```diff
@@ -5707,10 +5701,36 @@ define double @trig_preop_variable_args(double %arg0, i32 %arg1) {
 
 define double @trig_preop_constfold() {
 ; CHECK-LABEL: @trig_preop_constfold(
-; CHECK-NEXT:    [[VAL:%.*]] = call double @llvm.amdgcn.trig.preop.f64(double 3.454350e+02, i32 5)
-; CHECK-NEXT:    ret double [[VAL]]
+; CHECK-NEXT:    ret double 0x394A6EE06DB14ACC
```

Comment:
Do you have some way of checking that this matches the hardware result, for a few randomly chosen inputs?

Reply:
Yes, verified with an OpenCL program using the builtin to make the outputs.